### PR TITLE
Add error field to device.list message response

### PR DESCRIPTION
**Pull request by esdrasjnr**
What does this implement/fix? Explain your changes
---------------------------------------------------
This patch adds a `error` argument to `device.list` response

Where has this been tested?
---------------------------

**Operating System/Platform:** Ubuntu 18.04.4

**Go Version:** go1.13.8

### CODE REVIEW

**Comment by joaoaneto** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/thing/interactors/request_data.go  2
- pkg/thing/interactors/auth_thing.go  1
- pkg/thing/interactors/list_things.go  1
- pkg/thing/interactors/register_thing.go  3
         

Clones added
============
- pkg/thing/interactors/unregister_thing_test.go  2
         

Clones removed
==============
+ pkg/thing/interactors/update_schema_test.go  -9
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=5237370&bid=17166605)

**Comment by joaoaneto** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/thing/interactors/request_data.go  2
- pkg/thing/interactors/list_things.go  1
- pkg/thing/interactors/auth_thing.go  1
- pkg/thing/interactors/register_thing.go  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=5237370&bid=17166605)

File: pkg/thing/interactors/list_things.go [hunk 1]
```diff
@@ -10,10 +10,14 @@ func (i *ThingInteractor) List(authorization string) error {
...
 	if err != nil {
+		sendErr := i.clientPublisher.SendDevicesList(things, err)
+		if sendErr != nil {
+			return fmt.Errorf("error getting list of things: %v, %w", sendErr, err)
+		}
 		return fmt.Errorf("error getting list of things: %w", err)
 	}
 
-	err = i.clientPublisher.SendDevicesList(things)
+	err = i.clientPublisher.SendDevicesList(things, nil)
```
Comment (netoax) [inline pkg/thing/interactors/list_things.go]: Please, prefer passing `err` instead of `nil` directly to increase function cohesion.

**Review by netoax — CHANGES_REQUESTED** [review]
Good patch, thanks!
Just some adjustments to do.

_2fbc14e9299ffda2f74b9fef332de6fda921154a_:
```
In order to be compatible with the current AMQP API, this patch adds **an** error field when sending a device list response to the clients.
<link for the API documentation here>
```

File: pkg/thing/interactors/list_things.go (hunk 1 above)
Comment (esdrasjnr) [reply pkg/thing/interactors/list_things.go]: @joaoaneto You're right! I gonna make some changes

**Comment by joaoaneto** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/thing/interactors/list_things.go  1
- pkg/thing/interactors/request_data.go  2
- pkg/thing/interactors/auth_thing.go  1
- pkg/thing/interactors/register_thing.go  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=5237370&bid=17166605)

**Comment by joaoaneto** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/thing/interactors/request_data.go  2
- pkg/thing/interactors/register_thing.go  3
- pkg/thing/interactors/list_things.go  1
- pkg/thing/interactors/auth_thing.go  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=5237370&bid=17166605)

**Comment by joaoaneto** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/thing/interactors/request_data.go  2
- pkg/thing/interactors/list_things.go  1
- pkg/thing/interactors/register_thing.go  3
- pkg/thing/interactors/auth_thing.go  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=5237370&bid=17166605)

File: pkg/network/msg.go [hunk 2]
```diff
@@ -53,6 +53,7 @@ type DeviceAuthResponse struct {
 // DeviceListResponse represents the outgoing list devices command response
 type DeviceListResponse struct {
 	Things []*entities.Thing `json:"devices"`
+	ErrMsg *string           `json:"error"`
```
Comment (lcbm) [inline pkg/network/msg.go]: As per commit 2229801ed003112e864cdc4be095e9265f6994e9, this should be `Error` instead :grimacing:

**Review by lcbm — CHANGES_REQUESTED** [review]
Edit: Nevermind, this request change was by accident... LGTM :confetti_ball:

**Review by lcbm — APPROVED** [review]
LGTM :confetti_ball:

File: pkg/thing/interactors/list_things.go (hunk 1 above)
Comment (esdrasjnr) [reply pkg/thing/interactors/list_things.go]: Done!

**Comment by joaoaneto** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/thing/interactors/register_thing.go  3
- pkg/thing/interactors/list_things.go  1
- pkg/thing/interactors/request_data.go  2
- pkg/thing/interactors/auth_thing.go  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=5237370&bid=17166605)

**Review by netoax — APPROVED** [review]

File: pkg/thing/interactors/list_things.go [hunk 3]
```diff
@@ -9,13 +9,15 @@ func (i *ThingInteractor) List(authorization string) error {
 	}
 
 	things, err := i.thingProxy.List(authorization)
+	sendErr := i.clientPublisher.SendDevicesList(things, err)
```
Comment (netoax) [inline pkg/thing/interactors/list_things.go:12]: I think you're doing more than adding arguments to the list response, right?

**Review by netoax — CHANGES_REQUESTED** [review]

**Comment by joaoaneto** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/thing/interactors/register_thing.go  3
- pkg/thing/interactors/request_data.go  2
- pkg/thing/interactors/auth_thing.go  1
- pkg/thing/interactors/list_things.go  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=5237370&bid=17166605)

**Comment by netoax** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/thing/interactors/list_things.go  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=5237370&bid=17166605)

**Review by netoax — APPROVED** [review]

**Pull request merged**